### DISCUSSION TITLE
What is in the virtualenvs dirs, e.g. $HOME/.local/share/virtualenvs/Foo-1E7o8dGZ ?

### DISCUSSION THREAD

**Discussion by JDLH** [pypa/pipenv#5124] [Q&A] [answered]
I see that **pipenv** creates magical stuff in `$HOME/.local/share/virtualenvs/`, e.g. `Foo-1E7o8dGZ/`. Where is this magical stuff documented?  What is in there, what does it do? What kind of relationships are there from this stuff to the project directories to which they correspond? 

Does **pipenv** actually call **virtualenv**, and it is **virtualenv** which directly creates this stuff? If so, should I be asking in a this question in a virtualenv forum?

**Comment by matteius** [comment, marked as answer]
You can find the virtualenv for your current pipenv managed project by running `pipenv --venv`.   The virtualenv name is auto-generated from pipenv hence the `Foo-1E7o8dGZ`.   The virtualenv is a link to your version of Python for the project and all of the dependencies you require are installed and referenced from there for the project space.   Virtualenv provides isolation for your project dependencies.
